# Exercise

## Import Package

In [41]:
import pandas as pd
import numpy as np

## Import Data

In [2]:
imdb = pd.read_csv('IMDB-database-update.csv', encoding='latin1')

In [3]:
imdb.head()

movie_ID                               movie_title  duration  color  \
0         1                                    Avatar       178      1   
1         2  Pirates of the Caribbean: At World's End       169      1   
2         3                     The Dark Knight Rises       164      1   
3         4                              Spider-Man 3       156      1   
4         5        Batman v Superman: Dawn of Justice       183      1   

   title_year country language content_rating  budget_USD  gross_USD  \
0        2009     USA  English          PG-13   237000000  760505847   
1        2007     USA  English          PG-13   300000000  309404152   
2        2012     USA  English          PG-13   250000000  448130642   
3        2007     USA  English          PG-13   258000000  336530303   
4        2016     USA  English          PG-13   250000000  330249062   

       ...       actor_3_facebook?  actor_3_facebook_likes  \
0      ...                       1                     855   
1      ...                       1                    1000   
2      ...                       1                   23000   
3      ...                       1                    4000   
4      ...                       1                    2000   

  cast_total_facebook_likes movie_facebook?  movie_facebook_likes;  \
0                      4834               1                  33000   
1                     48350               0                      0   
2                    106759               1                 164000   
3                     46055               0                      0   
4                     24450               1                 197000   

   facenumber_in_poster num_voted_users  num_user_for_reviews  imdb_score  \
0                     0          886204                  3054         7.9   
1                     0          471220                  1238         7.1   
2                     0         1144337                  2701         8.5   
3                     0          383056                  1902         6.2   
4                     0          371639                  3018         6.9   

   imdb_top_250  
0             0  
1             0  
2             1  
3             0  
4             0  

[5 rows x 36 columns]

In [4]:
imdb.tail()

movie_ID         movie_title  duration  color  title_year country  \
3776      3777      New Year's Eve       113      1        2011     USA   
3777      3778       Boogie Nights       155      1        1997     USA   
3778      3779  Battle of the Year       110      1        2013     USA   
3779      3780          The Master       144      1        2012     USA   
3780      3781  500 Days of Summer        95      1        2009     USA   

     language content_rating  budget_USD  gross_USD      ...       \
3776  English          PG-13    56000000   54540525      ...        
3777  English              R    15000000   26384919      ...        
3778  English          PG-13    20000000    8888355      ...        
3779  English              R    32000000   16377274      ...        
3780  English          PG-13     7500000   32391374      ...        

      actor_3_facebook?  actor_3_facebook_likes cast_total_facebook_likes  \
3776                  1                     307                     23456   
3777                  1                     170                      3677   
3778                  1                     826                      3177   
3779                  1                      15                       427   
3780                  1                   11000                     54075   

     movie_facebook?  movie_facebook_likes;  facenumber_in_poster  \
3776               1                  20000                    15   
3777               0                      0                    15   
3778               0                      0                    19   
3779               1                  27000                    31   
3780               1                  40000                    43   

     num_voted_users  num_user_for_reviews  imdb_score  imdb_top_250  
3776           66579                   134         5.7             0  
3777          189032                   560         7.9             0  
3778            7894                    25         5.0             0  
3779          105144                   405         7.1             0  
3780          376600                   494         7.7             0  

[5 rows x 36 columns]

In [5]:
imdb.shape

(3781, 36)

In [6]:
imdb.columns

Index(['movie_ID', 'movie_title', 'duration', 'color', 'title_year', 'country',
       'language', 'content_rating', 'budget_USD', 'gross_USD', 'net_USD',
       'profitable', 'director_name', 'director_top20?', 'director_facebook?',
       'director_facebook_likes', 'actor_1_name', 'actor_top20?',
       'actor_1_facebook?', 'actor_1_facebook_likes', 'actor_2_name',
       'actor_top20?.1', 'actor_2_facebook?', 'actor_2_facebook_likes',
       'actor_3_name', 'actor_top20?.2', 'actor_3_facebook?',
       'actor_3_facebook_likes', 'cast_total_facebook_likes',
       'movie_facebook?', 'movie_facebook_likes;', 'facenumber_in_poster',
       'num_voted_users', 'num_user_for_reviews', 'imdb_score',
       'imdb_top_250'],
      dtype='object')

In [7]:
#Cleaning column names
imdb.columns = ['movie_ID', 'movie_title', 'duration', 'color', 'title_year', 'country',
       'language', 'content_rating', 'budget_USD', 'gross_USD', 'net_USD',
       'profitable', 'director_name', 'director_top20', 'director_facebook',
       'director_facebook_likes', 'actor_1_name', 'actor_top20',
       'actor_1_facebook', 'actor_1_facebook_likes', 'actor_2_name',
       'actor_top20.1', 'actor_2_facebook', 'actor_2_facebook_likes',
       'actor_3_name', 'actor_top20.2', 'actor_3_facebook',
       'actor_3_facebook_likes', 'cast_total_facebook_likes',
       'movie_facebook', 'movie_facebook_likes', 'facenumber_in_poster',
       'num_voted_users', 'num_user_for_reviews', 'imdb_score',
       'imdb_top_250']

### Check Duplicates

In [8]:
print(any(imdb['movie_title'].duplicated()))

True


Because there is duplicates in movie title column, drop the duplicates

## Cleaning Data

In [9]:
imdb.drop_duplicates(subset ="movie_title", 
                     keep = 'first', inplace = True) 

**1. How many rows and columns reflected on dataset file IMDB-database-update.csv?**

In [10]:
imdb.shape

(3679, 36)

There are 3679 rows and 36 columns

**2. What are the top 3 movies with highest nett profit in 2015?**

In [12]:
imdb[imdb['title_year']==2015].sort_values(by=['net_USD'], ascending=False).head(3)

movie_ID              movie_title  duration  color  title_year country  \
14          15           Jurassic World       124      1        2015     USA   
256        257                  Minions        91      1        2015     USA   
3384      3385  Avengers: Age of Ultron       141      1        2015     USA   

     language content_rating  budget_USD  gross_USD      ...       \
14    English          PG-13   150000000  652177271      ...        
256   English             PG    74000000  336029560      ...        
3384  English          PG-13   250000000  458991599      ...        

      actor_3_facebook  actor_3_facebook_likes cast_total_facebook_likes  \
14                   1                    1000                      8458   
256                  1                    1000                     13616   
3384                 1                   19000                     92000   

     movie_facebook  movie_facebook_likes  facenumber_in_poster  \
14                1                150000                     0   
256               1                 70000                     0   
3384              1                118000                     4   

     num_voted_users  num_user_for_reviews  imdb_score  imdb_top_250  
14            418214                  1290         7.0             0  
256           142403                   275         6.4             0  
3384          462669                  1117         7.5             0  

[3 rows x 36 columns]

**3. What are the top 3 movies with highest gross profit in 2016?**

In [13]:
imdb[imdb['title_year']==2016].sort_values(by=['gross_USD'], ascending=False).head(3)

movie_ID                 movie_title  duration  color  title_year  \
12         13  Captain America: Civil War       147      1        2016   
355       356                    Deadpool       108      1        2016   
36         37             The Jungle Book       106      1        2016   

    country language content_rating  budget_USD  gross_USD      ...       \
12      USA  English          PG-13   250000000  407197282      ...        
355     USA  English              R    58000000  363024263      ...        
36       UK  English             PG   175000000  362645141      ...        

     actor_3_facebook  actor_3_facebook_likes cast_total_facebook_likes  \
12                  1                   11000                     64798   
355                 1                     361                     18239   
36                  1                     591                     32921   

    movie_facebook  movie_facebook_likes  facenumber_in_poster  \
12               1                 72000                     0   
355              1                117000                     0   
36               1                 65000                     0   

    num_voted_users  num_user_for_reviews  imdb_score  imdb_top_250  
12           272670                  1022         8.2             1  
355          479047                  1058         8.1             1  
36           106072                   398         7.8             0  

[3 rows x 36 columns]

**4. How many non-english movies are in the dataset?**

In [18]:
imdb[imdb['language'] != 'English']['movie_title'].count()

152

**5. What are top 3 languages in the data set aside English (based on number of movies)?**

In [26]:
imdb[imdb['language'] != 'English']['language'].value_counts().head(3)

French      33
Spanish     23
Mandarin    13
Name: language, dtype: int64

**6. What are the top 3 countries with the highest nett profit in all dataset times?**

In [59]:
imdb.groupby(['country'])[['net_USD']].sum().sort_values(by=['net_USD'], ascending=False).head(3)

net_USD
country             
USA      50330300012
UK        1279724445
Canada     232040577

**7. What are the average budget of all movies in dataset in 2015?**

In [28]:
imdb[imdb['title_year'] == 2015]['budget_USD'].mean()

52102458.333333336

**8. What are the average budget of all movies in dataset in 2016?** 

In [29]:
imdb[imdb['title_year'] == 2016]['budget_USD'].mean()

66014754.09836066

**9. Which one is higher, average budget of all movies in dataset in 2015 or 2016?**

In [30]:
print(imdb[imdb['title_year'] == 2015]['budget_USD'].mean() < imdb[imdb['title_year'] == 2016]['budget_USD'].mean())

True


**10. What movie that resulted highest nett profit of all dataset times? How much did the movie make?** 

In [37]:
imdb.sort_values(by=['net_USD'], ascending=False).head(1)

movie_ID movie_title  duration  color  title_year country language  \
0         1      Avatar       178      1        2009     USA  English   

  content_rating  budget_USD  gross_USD      ...       actor_3_facebook  \
0          PG-13   237000000  760505847      ...                      1   

   actor_3_facebook_likes cast_total_facebook_likes movie_facebook  \
0                     855                      4834              1   

   movie_facebook_likes  facenumber_in_poster num_voted_users  \
0                 33000                     0          886204   

   num_user_for_reviews  imdb_score  imdb_top_250  
0                  3054         7.9             0  

[1 rows x 36 columns]

In [39]:
imdb[imdb['movie_title'] =='Avatar']['net_USD']

0    523505847
Name: net_USD, dtype: int64

# Use Case: High Rating Movies Criteria

Hints:
* Let’s set imdb_score benchmark as High-Rating movie with imdb_score > 7.5
* Create new variables “score_group” (“High” if imdb_score > 7.5 and “Low” if imdb <= 7.5)
* Analyze at least 5 variables from which you can start figure out criteria of high rated movie
* Analyse 5 more variables to strengthen criteria of high rated movies

In [60]:
new_cat_score = []

for row in imdb['imdb_score']:
    if row > 7.5:
        new_cat_score.append('High')
    else:
        new_cat_score.append('Low')

In [62]:
imdb['score_group'] = new_cat_score

In [63]:
imdb.head()

movie_ID                               movie_title  duration  color  \
0         1                                    Avatar       178      1   
1         2  Pirates of the Caribbean: At World's End       169      1   
2         3                     The Dark Knight Rises       164      1   
3         4                              Spider-Man 3       156      1   
4         5        Batman v Superman: Dawn of Justice       183      1   

   title_year country language content_rating  budget_USD  gross_USD  \
0        2009     USA  English          PG-13   237000000  760505847   
1        2007     USA  English          PG-13   300000000  309404152   
2        2012     USA  English          PG-13   250000000  448130642   
3        2007     USA  English          PG-13   258000000  336530303   
4        2016     USA  English          PG-13   250000000  330249062   

      ...       actor_3_facebook_likes  cast_total_facebook_likes  \
0     ...                          855                       4834   
1     ...                         1000                      48350   
2     ...                        23000                     106759   
3     ...                         4000                      46055   
4     ...                         2000                      24450   

  movie_facebook movie_facebook_likes  facenumber_in_poster  num_voted_users  \
0              1                33000                     0           886204   
1              0                    0                     0           471220   
2              1               164000                     0          1144337   
3              0                    0                     0           383056   
4              1               197000                     0           371639   

  num_user_for_reviews  imdb_score  imdb_top_250  score_group  
0                 3054         7.9             0         High  
1                 1238         7.1             0          Low  
2                 2701         8.5             1         High  
3                 1902         6.2             0          Low  
4                 3018         6.9             0          Low  

[5 rows x 37 columns]

In [65]:
imdb[imdb['score_group'] == 'High'].describe(include = 'all').transpose()

count unique                   top freq  \
movie_ID                    520    NaN                   NaN  NaN   
movie_title                 520    520        The Blind Side    1   
duration                    520    NaN                   NaN  NaN   
color                       520    NaN                   NaN  NaN   
title_year                  520    NaN                   NaN  NaN   
country                     520     26                   USA  360   
language                    520     20               English  454   
content_rating              520     10                     R  283   
budget_USD                  520    NaN                   NaN  NaN   
gross_USD                   520    NaN                   NaN  NaN   
net_USD                     520    NaN                   NaN  NaN   
profitable                  520    NaN                   NaN  NaN   
director_name               520    320      Steven Spielberg   14   
director_top20              520      3                     0  453   
director_facebook           520    NaN                   NaN  NaN   
director_facebook_likes     520    NaN                   NaN  NaN   
actor_1_name                520    335             Tom Hanks   11   
actor_top20                 520    NaN                   NaN  NaN   
actor_1_facebook            520    NaN                   NaN  NaN   
actor_1_facebook_likes      520    NaN                   NaN  NaN   
actor_2_name                520    452           Bill Murray    5   
actor_top20.1               520    NaN                   NaN  NaN   
actor_2_facebook            520    NaN                   NaN  NaN   
actor_2_facebook_likes      520    NaN                   NaN  NaN   
actor_3_name                520    485  Alexandra Maria Lara    3   
actor_top20.2               520    NaN                   NaN  NaN   
actor_3_facebook            520    NaN                   NaN  NaN   
actor_3_facebook_likes      520    NaN                   NaN  NaN   
cast_total_facebook_likes   520    NaN                   NaN  NaN   
movie_facebook              520    NaN                   NaN  NaN   
movie_facebook_likes        520    NaN                   NaN  NaN   
facenumber_in_poster        520    NaN                   NaN  NaN   
num_voted_users             520    NaN                   NaN  NaN   
num_user_for_reviews        520    NaN                   NaN  NaN   
imdb_score                  520    NaN                   NaN  NaN   
imdb_top_250                520    NaN                   NaN  NaN   
score_group                 520      1                  High  520   

                                  mean          std          min          25%  \
movie_ID                       1709.73      1022.54            1          894   
movie_title                        NaN          NaN          NaN          NaN   
duration                       125.777      31.6446           37          105   
color                         0.917308     0.275682            0            1   
title_year                     1999.14      14.5989         1927         1994   
country                            NaN          NaN          NaN          NaN   
language                           NaN          NaN          NaN          NaN   
content_rating                     NaN          NaN          NaN          NaN   
budget_USD                 5.30537e+07   2.2007e+08        40000        7e+06   
gross_USD                  8.44989e+07  1.07306e+08         1111  1.09872e+07   
net_USD                    3.14452e+07  2.34635e+08 -4.19979e+09      -872144   
profitable                    0.717308     0.450742            0            0   
director_name                      NaN          NaN          NaN          NaN   
director_top20                     NaN          NaN          NaN          NaN   
director_facebook             0.701923     0.457854            0            0   
director_facebook_likes        2252.62      5391.19            0            0   
actor_1_name              

In [66]:
imdb[imdb['score_group'] == 'Low'].describe(include = 'all').transpose()

count unique             top  freq         mean  \
movie_ID                   3159    NaN             NaN   NaN      1917.16   
movie_title                3159   3159   The Producers     1          NaN   
duration                   3159    NaN             NaN   NaN       107.23   
color                      3159    NaN             NaN   NaN     0.975309   
title_year                 3159    NaN             NaN   NaN      2003.64   
country                    3159     44             USA  2567          NaN   
language                   3159     30         English  3073          NaN   
content_rating             3159     12               R  1395          NaN   
budget_USD                 3159    NaN             NaN   NaN  4.39565e+07   
gross_USD                  3159    NaN             NaN   NaN  4.54381e+07   
net_USD                    3159    NaN             NaN   NaN  1.48162e+06   
profitable                 3159    NaN             NaN   NaN     0.498575   
director_name              3159   1585    Renny Harlin    15          NaN   
director_top20             3159      2               0  2941          NaN   
director_facebook          3159    NaN             NaN   NaN     0.850902   
director_facebook_likes    3159    NaN             NaN   NaN      569.874   
actor_1_name               3159   1296  Robert De Niro    34          NaN   
actor_top20                3159    NaN             NaN   NaN     0.123457   
actor_1_facebook           3159    NaN             NaN   NaN     0.999367   
actor_1_facebook_likes     3159    NaN             NaN   NaN      7387.99   
actor_2_name               3159   1970  Morgan Freeman    14          NaN   
actor_top20.1              3159    NaN             NaN   NaN    0.0237417   
actor_2_facebook           3159    NaN             NaN   NaN     0.997151   
actor_2_facebook_likes     3159    NaN             NaN   NaN      1850.74   
actor_3_name               3159   2308    Steve Coogan     8          NaN   
actor_top20.2              3159    NaN             NaN   NaN   0.00348211   
actor_3_facebook           3159    NaN             NaN   NaN     0.993036   
actor_3_facebook_likes     3159    NaN             NaN   NaN      711.297   
cast_total_facebook_likes  3159    NaN             NaN   NaN      10916.9   
movie_facebook             3159    NaN             NaN   NaN     0.514087   
movie_facebook_likes       3159    NaN             NaN   NaN      6399.19   
facenumber_in_poster       3159    NaN             NaN   NaN      1.43431   
num_voted_users            3159    NaN             NaN   NaN      70880.1   
num_user_for_reviews       3159    NaN             NaN   NaN      266.269   
imdb_score                 3159    NaN             NaN   NaN      6.21187   
imdb_top_250               3159    NaN             NaN   NaN            0   
score_group                3159      1             Low  3159          NaN   

                                   std          min          25%          50%  \
movie_ID                       1103.29            2          947         1932   
movie_title                        NaN          NaN          NaN          NaN   
duration                       19.1882           45           95          104   
color                         0.155207            0            1            1   
title_year                      8.6325         1929         1999         2005   
country                            NaN          NaN          NaN          NaN   
language                           NaN          NaN          NaN          NaN   
content_rating                     NaN          NaN          NaN          NaN   
budget_USD                 2.29443e+08          218        1e+07      2.5e+07   
gross_USD                  5.85581e+07          162  6.84685e+06  2.64213e+07   
net_USD                    2.31321e+08 -1.22133e+10 -1.16837e+07       -15622   
profitable                    0.500077            0            0            0   
director_name                      NaN        

print(imdb[imdb['score_group'] == 'Low']['director_facebook_likes'].mean() < imdb[imdb['score_group'] == 'High']['director_facebook_likes'].mean())

In [70]:
print(imdb[imdb['score_group'] == 'Low']['actor_1_facebook_likes'].mean() < imdb[imdb['score_group'] == 'High']['actor_1_facebook_likes'].mean())

True


In [71]:
print(imdb[imdb['score_group'] == 'Low']['actor_2_facebook_likes'].mean() < imdb[imdb['score_group'] == 'High']['actor_2_facebook_likes'].mean())

True


In [72]:
print(imdb[imdb['score_group'] == 'Low']['actor_3_facebook_likes'].mean() < imdb[imdb['score_group'] == 'High']['actor_3_facebook_likes'].mean())

True


In [73]:
print(imdb[imdb['score_group'] == 'Low']['movie_facebook_likes'].mean() < imdb[imdb['score_group'] == 'High']['movie_facebook_likes'].mean())

True


it seems that number of likes(director, actors, cast, and movies) overall would like to impact imdb score, but this need to further proving